### DISCUSSION TITLE
[feature request] adding a logarithmic colormap normalization to drawing methods

### DISCUSSION THREAD

**Discussion by RiccardoBasilone** [networkx/networkx#7438] [Ideas]
Hi there,
I've been using networkx for a few months now and i find it works very well. 
I was wondering if it would be possible to implement colormaps that aren't mpl.colors.Normalize into the drawing methods. Sometimes it's useful to plot node or edge attributes normalized logarithmically, but from what i see in the documentation for draw_networkx_edges, you can pass a color map but can't choose the type of normalization.
In the source code for draw_networkx_edges  i see that there is a line
`color_normal = mpl.colors.Normalize(vmin=edge_vmin, vmax=edge_vmax)`
maybe making this a parameter in calling the function one could choose any of the normalizations supported by matplotlib, such as LogNorm?
Thanks for your consideration.

**Comment by dschult** [comment]
Thanks for the feedback and suggestion.

I think you should be able to provide edge_color directly to get the transformation you want. If you set up the colors using `LogNorm` they should work (and not be overwritten by the current code). But there may be subtleties I am not aware of. Can you try something like:

```python
    color_normal = mpl.colors.LogNorm(vmin=edge_vmin, vmax=edge_vmax)
    ecolor = [edge_cmap(color_normal(e)) for e in edge_color]

    nx.draw_networkx_edges(G, pos=pos, edge_color=ecolor) 
```

**Reply by RiccardoBasilone** [reply]
Hi again, 
sorry I'm just now getting back to this. I tested your code on my graph and as you anticipated, it works. However, results are produced in the [0,1] range, whereas I would like to produce values within the actual range of values, but i want the colors to be spaced logarithmically.
attached is a working example (.ipynb and a graph saved in .gml) to reproduce my situation, but i'll include screenshots of what i mean here:
![Screenshot 2024-05-09 203629](https://github.com/networkx/networkx/assets/123843611/b9d66db4-cfd0-4c55-b082-d3b29a8563c4)

![Screenshot 2024-05-09 170055](https://github.com/networkx/networkx/assets/123843611/c8f4985e-38c3-4090-b72f-6dc9ad5247e3)

[working_example.zip](https://github.com/networkx/networkx/files/15266281/working_example.zip)

I believe that if nx.draw_network_edges had a norm parameter, like plt.scatter does, the correct result could be achieved.
That being said, I think the workaround you provided is already a good start, and I appreciate your help.
